### PR TITLE
replaced random2 internals with std::random

### LINKED ISSUE #118: Replace random2 with std::random implementation

### PR DESCRIPTION
**Pull request by JensWehner**
https://github.com/votca/tools/issues/118  and https://github.com/votca/xtp/pull/293

Fix #118

### CODE REVIEW

File: src/tests/test_random2.cc [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  int seed = 1;
+  random.init(seed);
+  std::vector<double> results;
+  int number = 1e5;
+  results.reserve(number);
+  for (int i = 0; i < number; i++) {
+    results.push_back(random.rand_uniform());
+  }
+
+  // average should be close to 25
```
Comment (junghans) [inline src/tests/test_random2.cc]: 0.5 I guess?

Comment (JensWehner) [reply src/tests/test_random2.cc]: yep

**Comment by junghans** [comment]
2 questions: why aren’t we naming it “Random” without the “2”.  And why do we need this class at all?

**Comment by JensWehner** [comment]
Good questions. a) history, we can rename it, but I was to lazy. We should probably rename it. b) We could also not use it, I guess this gives it an even easier interface.

**Comment by junghans** [comment]
Then, I would say let's switch xtp to `std::random` and kill this class with fire.

**Comment by JensWehner** [comment]
If we delete the class here I would just move it to xtp, because I need exactly this interface. I do not care it is up to you.

**Comment by junghans** [comment]
Then, let's leave it here, but rename is to `Random`.

**Comment by JensWehner** [comment]
okay. I ll do that

**Pull request merged**